opsmngr/server_usage.go

```diff
@@ -0,0 +1,299 @@
+		return nil, nil, err
+	}
+	req, err := s.Client.NewRequest(ctx, http.MethodGet, path, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	root := new(HostAssignments)
+	resp, err := s.Client.Do(ctx, req, root)
+
+	return root, resp, err
+}
+
+// OrganizationHostAssignments retrieves all host assignments in a organization.
+//
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/usage/list-all-host-assignments-in-one-organization/
+func (s *ServerUsageServiceOp) OrganizationHostAssignments(ctx context.Context, orgID string, options *ServerTypeOptions) (*HostAssignments, *atlas.Response, error) {
+	if orgID == "" {
+		return nil, nil, atlas.NewArgError("orgID", "must be set")
+	}
+
+	basePath := fmt.Sprintf(serverUsageOrgPath, orgID)
+	path := fmt.Sprintf("%s/%s", basePath, "hosts")
+	path, err := setQueryParams(path, options)
+	if err != nil {
+		return nil, nil, err
+	}
+	req, err := s.Client.NewRequest(ctx, http.MethodGet, path, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	root := new(HostAssignments)
+	resp, err := s.Client.Do(ctx, req, root)
+
+	return root, resp, err
+}
+
+// GetServerTypeProject retrieves one default server type for one project.
+//
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/usage/get-default-server-type-for-one-project/
+func (s *ServerUsageServiceOp) GetServerTypeProject(ctx context.Context, groupID string) (*ServerType, *atlas.Response, error) {
+	if groupID == "" {
+		return nil, nil, atlas.NewArgError("groupID", "must be set")
+	}
+
+	basePath := fmt.Sprintf(serverUsageGroupPath, groupID)
+	path := fmt.Sprintf("%s/%s", basePath, "defaultServerType")
+
+	req, err := s.Client.NewRequest(ctx, http.MethodGet, path, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	root := new(ServerType)
+	resp, err := s.Client.Do(ctx, req, root)
+
+	return root, resp, err
+}
+
+// GetServerTypeOrganization retrieves one default server type for one organization.
+//
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/usage/get-default-server-type-for-one-organization/
+func (s *ServerUsageServiceOp) GetServerTypeOrganization(ctx context.Context, orgID string) (*ServerType, *atlas.Response, error) {
+	if orgID == "" {
+		return nil, nil, atlas.NewArgError("orgID", "must be set")
+	}
+
+	basePath := fmt.Sprintf(serverUsageOrgPath, orgID)
+	path := fmt.Sprintf("%s/%s", basePath, "defaultServerType")
+
+	req, err := s.Client.NewRequest(ctx, http.MethodGet, path, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	root := new(ServerType)
+	resp, err := s.Client.Do(ctx, req, root)
+
+	return root, resp, err
+}
+
+// UpdateProjectServerType update the default server type for one project.
+//
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/usage/update-default-server-type-for-one-project/
+func (s *ServerUsageServiceOp) UpdateProjectServerType(ctx context.Context, groupID string, serverType *HostAssignment) (*atlas.Response, error) {
```

Comment:
I'm looking at the code and this change doesn't seem right

Reply:
what the endpoint needs is 
```
			   "serverType":{
				  "name":"RAM_POOL",
				  "label":"RAM Pool"
			   }
```

which is inside HostAssignment. The serverType struct only has name and label

Reply:
ok, I see what you mean, we miss the key here
can we then add a new type? something like
```go
type ServerTypeRequest struct {	
	ServerType   *ServerType              `json:"serverType,omitempty"`
}
```

it's a bit confusing using that big struct with so many fields to be ignore